5.2 Ex/Cup of join.py

```diff
@@ -0,0 +1,48 @@
+"""Cup of join
+Receives from user lists and separator, return the lists into one,
+separate them with the wanted separator (or '-' if separator doesn't receive)
+and print the list.
+"""
+
+from functools import reduce
+
+
+def join(*tuple_of_lists: tuple, sep: str = '-') -> list or None:
```

Comment:
nice use of tuple here! but maybe a more correct use would be to define the type as iterable (read about abc.iterable and typing.iterable), because you actually dont mind which kind of iterable is it

Reply:
Hey Tal,
I think I disagree with you because (for example) I can't add two sets.
Waiting for your answer at slack.